services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -746,6 +765,20 @@ async def check_delete_project_permission(self, user_id: int, project_uuid: str)
                 )
                 _check_project_permissions(project, user_id, user_groups, "delete")
 
+    async def check_project_has_only_one_product(self, project_uuid: ProjectID) -> None:
+        async with self.engine.acquire() as conn:
+            num_products_linked_to_project = await conn.scalar(
+                sa.select(func.count())
+                .select_from(projects_to_products)
+                .where(projects_to_products.c.project_uuid == f"{project_uuid}")
+            )
+            assert num_products_linked_to_project  # nosec
+        if num_products_linked_to_project > 1:
+            raise ProjectDeleteError(
```

Comment:
Do i understand this right? Projects associated to multiple products cannot be deleted using the API

I wonder if this would lead to too many "manual interventions"

why didn't you decide to "unlink" this product and skip the project deletion instead of just error deletion?

Reply:
This is in agreement with @odeimaiz :
- we decided that this precise use-case where we have a project in 2 products at the same time does not exist now and its chances to ever come is small -> therefore we do not treat it for now
- nevertheless in case it would be done this way (which would by the way need a manual intervention already to set it up), at least it would not be possible to delete the project in one product to prevent some unwanted deletion.
- reduce time to develop by not implementing something that might never be necessary

Reply:
I see. The initial design was actually the opposite, i.e. encouraging the idea of sharing projects through products. But i understand that the specs evolve.

Can you please add those points above in the code so we have the context of this decision?